PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,650 @@
+                     observerMode: Bool,
+                     subscriberAttributes subscriberAttributesByKey: SubscriberAttributeDict?,
+                     completion: @escaping BackendPurchaserInfoResponseHandler) {
+    // swiftlint:enable function_parameter_count
+        let fetchToken = receiptData.base64EncodedString(options: .lineLength64Characters)
+        var body: [String: AnyObject] = [
+            "fetch_token": fetchToken as NSString,
+            "app_user_id": appUserID as NSString,
+            "is_restore": NSNumber(value: isRestore),
+            "observer_mode": NSNumber(value: observerMode),
+        ]
+
+        let cacheKey =
+        """
+        \(appUserID)-\(isRestore)-\(fetchToken)-\(productInfo?.cacheKey ?? "")
+        -\(offeringIdentifier ?? "")-\(observerMode)-\(subscriberAttributesByKey?.debugDescription ?? "")"
+        """
+
+        if add(callback: completion, key: cacheKey) {
+            return
+        }
+
+        if let productInfo = productInfo {
+            body.merge(productInfo.asDictionary()) { _, new in new }
+        }
+
+        if let subscriberAttributesByKey = subscriberAttributesByKey {
+
+            let attributesInBackendFormat = subscriberAttributesToDict(subscriberAttributes: subscriberAttributesByKey)
+            body["attributes"] = attributesInBackendFormat as AnyObject
+        }
+
+        if let offeringIdentifier = offeringIdentifier {
+            body["presented_offering_identifier"] = offeringIdentifier as NSString
+        }
+
+        httpClient.performPOSTRequest(serially: true,
+                                      path: "/receipts",
+                                      requestBody: body,
+                                      headers: authHeaders) { statusCode, response, error in
+            let callbacks = self.getPurchaserInfoCallbacksAndClearCache(forKey: cacheKey)
+            for callback in callbacks {
+                self.handle(purchaserInfoResponse: response, statusCode: statusCode, error: error, completion: callback)
+            }
+        }
+    }
+
+    @objc
+    public func getSubscriberData(appUserID: String, completion: @escaping BackendPurchaserInfoResponseHandler) {
+        guard let appUserID = try? escapedAppUserID(appUserID: appUserID) else {
+            completion(nil, ErrorUtils.missingAppUserIDError())
+            return
+        }
+
+        let path = "/subscribers/\(appUserID)"
+
+        if add(callback: completion, key: path) {
+            return
+        }
+
+        httpClient.performGETRequest(serially: true, path: path, headers: authHeaders) { statusCode, response, error in
+            for completion in self.getPurchaserInfoCallbacksAndClearCache(forKey: path) {
+                print("Hi")
+                self.handle(purchaserInfoResponse: response,
+                            statusCode: statusCode,
+                            error: error,
+                            completion: completion)
```

Comment:
note, just for the sake of discussion: this might be a case for a `weak self`:

If we're getting subscriber info, but the `Backend` instance has been dealloc'ed, it's likely that the SDK has been re-configured, and it might have a different user. 
In a case like this, we might not want to overwrite the data in cache with a new value from a potentially old user. 

However, I think deciding that shouldn't be up to the `Backend` class, but rather the `IdentityManager` or even `PurchaserInfoManager`. So I think the strong self here is fine.

Reply:
This is a great catch. I'm going back through the this class and seeing where it makes sense for `weak self`. I actually had a single failing test related to re-configuring the SDK, and switching it to `weak self` fixes it.